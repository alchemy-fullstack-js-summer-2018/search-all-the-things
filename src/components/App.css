:local(.app) {

  h1 {
    text-shadow: 2px 2px #000000;
  }

  ul {
    list-style: none;
    display: flex;
    flex-direction: row;
    align-items: center;
    flex-flow: row wrap;
  }

}