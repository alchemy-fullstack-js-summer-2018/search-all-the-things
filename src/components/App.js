import React, { Component } from 'react';
import Header from './Header';
import Books from './books/Books';
import styles from './App.css';
import { search } from '../services/booksApi';

class App extends Component {

  state = {
    data: []
  };

  handleSearch = (term) => {
    console.log(`You searched for ${term.search}`);

    search(term)
      .then(results => this.setState({ data: results }));
    console.log('****AFTER SEARCH*****');
  };

  render() {

    return (
      <main className={styles.app}>
        <h1>Alchemy Code Lab Library</h1>
        <header>
          <Header onSearch={this.handleSearch}/>
        </header>

        <section>
          <Books />
        </section>
      </main>
    );
  }
}

export default App;