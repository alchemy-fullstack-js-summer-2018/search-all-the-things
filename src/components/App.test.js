import * as React from 'react';
import { shallow, configure  } from 'enzyme';
import App from './App';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('App testing', () => {
    
  it('App renders as designed', () => {
    const wrapper = shallow(<App/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

});