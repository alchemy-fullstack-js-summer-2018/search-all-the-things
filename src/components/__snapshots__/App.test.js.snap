// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App testing App renders as designed 1`] = `
<BrowserRouter>
  <div>
    <header>
      <Header />
    </header>
    <main>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/favorites"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/search"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/movies/:id"
        />
        <Redirect
          push={false}
          to="/"
        />
      </Switch>
    </main>
  </div>
</BrowserRouter>
`;
